src/NUnitFramework/tests/Assertions/WarningTests.cs

```diff
@@ -283,9 +284,19 @@ public static void StackTracesAreFiltered(string methodName, int maxLineCount)
             }
 
             var warningStackTrace = result.AssertionResults[0].StackTrace;
-            var lineCount = warningStackTrace.Split(new[] { '\n' }, StringSplitOptions.RemoveEmptyEntries).Length;
+            var lines = warningStackTrace.Split(new[] { '\n' }, StringSplitOptions.RemoveEmptyEntries);
 
-            Assert.That(lineCount, Is.LessThanOrEqualTo(maxLineCount));
+            if (maxLineCount < lines.Length)
+            {
+                Assert.Fail(
+                     $"Expected the number of lines to be no more than {maxLineCount}, but it was {lines.Length}:" + Environment.NewLine
+                     + Environment.NewLine
+                     + string.Concat(lines.Select((line, i) => $" {i + 1}. {line.Trim()}" + Environment.NewLine))
+                     + "(end)");
```

Comment:
Why not just use three- or four-argument form of assert?

Reply:
Hey, I'm not familiar with that. What would that look like?

Reply:
You can do

```C#
Assert.That(lines.Length,
            Is.LessThanOrEqualTo(maxLineCount), 
            $"Expected the number of lines to be no more than {maxLineCount}, but it was {lines.Length}:" 
               + Environment.NewLine
               + Environment.NewLine
               + string.Concat(lines.Select((line, i) => $" {i + 1}. {line.Trim()}" + Environment.NewLine))
               + "(end)");
```

The message is constructed before the call. Usually efficiency is not a great concern in assert messages but if you are concerned you can use a lambda returning a string.

Reply:
Ah, using `Assert.That`. I chose not to use it because I wanted the stack after the expected number of lines message and didn't want the message repeated in two places.